downloads/views.py

```diff
@@ -48,20 +69,60 @@ def get_presigned_url(
     )
 
 
-def stat_object(job_id: str, task_id: str, node_id: str, timestamp_and_filename: str):
+def stat_osn_object(item: Item):
     client = get_minio_client()
-    object_name = get_object_name(job_id, task_id, node_id, timestamp_and_filename)
+    object_name = get_osn_object_name(item)
     return client.stat_object(
         bucket_name=settings.S3_BUCKET_NAME,
         object_name=object_name,
     )
 
 
-def get_timestamp_for_timestamp_and_filename_string(s: str) -> datetime:
-    s = s.split("-", maxsplit=1)[0]
-    nanos = float(s)
-    timestamp = nanos / 1e9
-    return datetime.fromtimestamp(timestamp, tz=timezone.utc)
+def object_exists_in_osn(item: Item):
+    client = get_minio_client()
+    object_name = get_osn_object_name(item)
+
+    try:
+        client.stat_object(
+            bucket_name=settings.S3_BUCKET_NAME,
+            object_name=object_name,
+        )
+        return True
+    except S3Error:
+        return False
+
+
+def get_pelican_path(item: Item):
+    return f"/node-data/{item.job_id}/{item.task_id}/{item.node_id}/{item.timestamp_and_filename}"
+
+
+def get_pelican_authz_url(item: Item):
+    path = get_pelican_path(item)
+
+    key = Path(settings.PELICAN_KEY).read_bytes()
+
+    token = SciToken(
+        key=key,
+        algorithm=settings.PELICAN_ALGORITHM,
+        key_id=settings.PELICAN_KEY_ID,
+    )
+    token["sub"] = "test"
+    token["aud"] = "ANY"
+    token["ver"] = "scitoken:2.0"
+    token["scope"] = f"read:{path}"
+
+    authz = token.serialize(
+        issuer=settings.PELICAN_ISSUER,
```

Comment:
Same goes for here, but again it's not important since we won't be changing our issuer much

Reply:
Thanks!